### ISSUE TITLE
onHide is shown when a dialog is created

### ISSUE THREAD

**Issue by mmikeyy** [primefaces/primeng#3051]
**I'm submitting a ...**  (check one with "x")
```
[ x] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Plunkr Case (Bug Reports)**
impossible. please fix the plunkr provided to demonstrate errors. 1) it is based on 4.0.0 which is dated; 2) it does not include animations, so dialogs are a problem.

**Current behavior**
Dialog component fires onHide event upon creation when the dialog is created in the hidden state.

**Expected behavior**
onHide should only fire when a visible dialog is closed

**Minimal reproduction of the problem with instructions**
create a dialog with [visible]="false" (onHide)="fireEvent()" and see the event being fired.

**What is the motivation / use case for changing the behavior?**

The onHide event is used to signal the end of a process happening in the dialog component, but the process never even started.

**Please tell us about your environment:**
irrelevant

* **Angular version:** latest
* **PrimeNG version:** 4.0.3

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

 
* **Language:** typescript 2.3.2

* **Node (for AoT issues):** `node --version` =

**Comment by mtroszczanczuk** [comment]
The same problem is happening to me! In my case I am building a dialog to prevent the page exchange and when canceling (which would be the action of leaving the page) the user would be directed off the page. In this case I can not even enter the page, because the onHide event is being triggered as soon as I enter the page, without even the dialog being displayed. Comparing the code of version 4.0.0 with that of 4.0.3 the difference I see is inside the set of visible, where in the if of the container that in version 4.0.0 was initialized only in the ngAfterViewInit event and in version 4.0.3 it already Tests the content of the content thus allowing premature firing of the hide event.

**Comment by cagataycivici** [comment]
This should be fixed in recent PrimeNG release, please try with it and if the issue persists drop a comment with along with a test case based on plunkr below and we'll review again.

http://plnkr.co/edit/Wj39h1?p=preview

**Comment by mmikeyy** [comment]
This is not fixed in 4.1 rc3.

See plunkr here: [](http://plnkr.co/edit/6YjZrJMht7ooIi2tUkaW?p=preview)

This is a major annoyance.
